NTupleAnalyzerHH_nanoAOD.py

```diff
@@ -893,37 +973,37 @@ def LeptonsAndJetsFromHH(T):
 	recoelectroninds = []
 	onShellZMu=False
 	onShellZEle=False
+	matchedrecomuons=[]
+	matchedrecoelectrons=[]
+	emptyvector = TLorentzVector()
+	emptyvector.SetPtEtaPhiM(0,0,0,0)
 
+###################### genMatching Muons ##########################
 	for n in range(len(T.Muon_pt)):
 		m = TLorentzVector()
-		m.SetPtEtaPhiM(T.Muon_pt[n],T.Muon_eta[n],T.Muon_phi[n],0)
+		m.SetPtEtaPhiM(T.Muon_pt[n],T.Muon_eta[n],T.Muon_phi[n],T.Muon_mass[n])
 		muons.append(m)
-	for n in range(T.nGenPart):
-		pdg = T.GenPart_pdgId[n]
-		if pdg not in [13,-13]:
-			continue
-		motherIndex = T.GenPart_genPartIdxMother[n]
-
-		motherid = 0
-		if motherIndex>-1:
-			motherid = T.GenPart_pdgId[motherIndex]
-		#if v=='': print 'pdgId:',pdg,'mother index:',T.GenParticleMotherIndex[n],'muStatus:',T.GenParticleStatus[n],'mother pdgId:',T.GenParticlePdgId[motherIndex], 'mu pt', T.GenParticlePt[n]
-		if motherid not in [23,-23,25,-25]:
-			continue
-		if motherid in [23,-23]: onShellZMu=True
-		#elif v=='': print 'Muon not from Z???  muStatus:',T.GenParticleStatus[n],'  Muparent:',motherid
-
-		# AH: here we keep also gmuon status 23 and status 1, of which mother 25. But we do not keep muon status 1 of which mother is 13, -13
-		m = TLorentzVector()
-		m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],0.0)
-		genmuons.append(m)
-		#if v=='': print ' gen muon pt ', m.Pt(), 'eta', m.Eta()
 
-	matchedrecomuons=[]
-	matchedrecoelectrons=[]
-	emptyvector = TLorentzVector()
-	emptyvector.SetPtEtaPhiM(0,0,0,0)
-	#print 'number of genmuons:',len(genmuons)
+	if isdata==False:
+
+		for n in range(T.nGenPart):
+			pdg = T.GenPart_pdgId[n]
+			if pdg not in [13,-13]:
+				continue
+			motherIndex = T.GenPart_genPartIdxMother[n]
+
+			motherid = 0
+			if motherIndex>-1:
+				motherid = T.GenPart_pdgId[motherIndex]
+			if motherid not in [23,-23,25,-25]:
+				continue
+			if motherid in [23,-23]: onShellZMu=True
+
+			# AH: here we keep also gmuon status 23 and status 1, of which mother 25. But we do not keep muon status 1 of which mother is 13, -13
+			m = TLorentzVector()
+			m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],T.GenPart_mass[n])
```

Comment:
This looks like a problem in the indentation compared to the old version, please check.